CoreUtilities/test/unit/src/au/gov/asd/tac/constellation/utilities/xml/XmlUtilitiesNGTest.java

```diff
@@ -0,0 +1,907 @@
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+import org.testng.reporters.Files;
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+/**
+ *
+ * @author mmattner
+ */
+public class XmlUtilitiesNGTest {
+    
+    private static final String OUTPUT_FILE = "testOutputFile.xml";
+    private static final String XML_HDR = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>";
+    
+    public XmlUtilitiesNGTest() {
+    }
+
+    @BeforeClass
+    public static void setUpClass() throws Exception {
+    }
+
+    @AfterClass
+    public static void tearDownClass() throws Exception {
+    }
+
+    @BeforeMethod
+    public void setUpMethod() throws Exception {
+//        }
+    }
+
+    @AfterMethod
+    public void tearDownMethod() throws Exception {
+    }
+
+    /**
+     * Test of newDocument method, of class XmlUtilities. This effectively creates a new empty 
+     * document.
+     */
+    @Test
+    public void testNewDocument() {
+        System.out.println("testNewDocument");
+        XmlUtilities instance = new XmlUtilities();
+       
+        Document expResult = null;
+        try {
+            expResult = DocumentBuilderFactory.newInstance().newDocumentBuilder().newDocument(); 
+        } catch (ParserConfigurationException ex) {
+            fail("Exception thrown creating sample data");
+        }
+        Document result = instance.newDocument();
+        assertTrue(result.isEqualNode(expResult));
+    }
+
+    /**
+     * Test of write method, of class XmlUtilities.
+     */
+    @Test
+    public void testWrite_Document() throws Exception {
+        System.out.println("testWrite_Document");
+        XmlUtilities instance = new XmlUtilities();
+        
+        try {
+            String expectedStr = XML_HDR + "<parent>\n"
+                    + "  <child1>child1_value</child1>\n"
+                    + "  <child2>child2_value</child2>\n"
+                    + "  <child3>child3a_value</child3>\n"
+                    + "  <child3>child3b_value</child3>\n"
+                    + "  <child4><child4.1>child4.1</child4.1></child4>\n"
+                    + "</parent>\n";
+
+            String testFile = XmlUtilitiesNGTest.class.getResource("resources/testWrite_Document.xml").getPath();
+            DocumentBuilderFactory docBuilderFactory = DocumentBuilderFactory.newInstance();
+            DocumentBuilder docBuilder = docBuilderFactory.newDocumentBuilder();
+            Document document = docBuilder.parse(new File(testFile));
+
+            byte[] results = instance.write(document);
+            String resultStr = (new String(results));
+            assertTrue(resultStr.replaceAll("\\s+","").equals(expectedStr.replaceAll("\\s+","")));
+            
+        } catch (Exception ex) {
+            fail("The test threw an unexpected exception.");
+        }
```

Comment:
similar here (and I'm guessing for other tests as well)

Reply:
done